File: EXAMEN VIERNES/sistema-reserva-libros/src/main/java/com/diegoAlvarez/sistema_reserva_libros/service/LibroService.java
package com.diegoAlvarez.sistema_reserva_libros.service;

import com.diegoAlvarez.sistema_reserva_libros.dto.LibroDTO;
import com.diegoAlvarez.sistema_reserva_libros.model.Libro;

import java.util.List;

public interface LibroService {
    Libro crearLibro(LibroDTO libroDTO);
    List<Libro> listarLibros();
    List<Libro> listarLibrosDisponibles();
    Libro obtenerLibroPorId(Long id);
}

File: EXAMEN VIERNES/FRONTEND/sistema-reserva-libros-frontend/src/app/services/libro.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/api-response.model';
import { Libro, LibroDTO } from '../models/libro.model';

@Injectable({
  providedIn: 'root'
})
export class LibroService {
  private readonly url = 'http://localhost:8080/api/libros';

  constructor(private readonly http: HttpClient) { }

  listarLibros(): Observable<ApiResponse<Libro[]>> {
    return this.http.get<ApiResponse<Libro[]>>(this.url);
  }

  listarLibrosDisponibles(): Observable<ApiResponse<Libro[]>> {
    return this.http.get<ApiResponse<Libro[]>>(`${this.url}/disponibles`);
  }

  crearLibro(libro: LibroDTO): Observable<ApiResponse<Libro>> {
    return this.http.post<ApiResponse<Libro>>(this.url, libro);
  }
}